<?xml version="1.0"?>

<launch>
	<!-- launch Gazebo world -->
	<include file="$(find go_thesis)/launch/ocean_waves.launch"/>

	<!-- launch AUV model -->
	<include file="$(find desistek_saga_description)/launch/upload_desistek_saga.launch">
	    <arg name="x" value="4.0"/>
    	<arg name="y" value="4.0"/>
    	<arg name="z" value="-5.0"/>
    	<arg name="yaw" value="0.0"/>
    </include>
	
	<!-- launch ASV model -->
	<group ns="heron">
		<param name="tf_prefix" value="heron_tf"/>
		<include file="$(find heron_gazebo)/launch/heron_sim.launch"/>
	</group>

    <!-- launch AUV Joystick controler-->
    <include file="$(find desistek_saga_control)/launch/start_cascaded_pid_with_teleop.launch">
        <arg name="joy_id" value="0"/>
    </include>

	<!-- cmd Heron -->
	<node pkg="go_thesis" type="heron_thrustersManager.py" name="heron_thusters_manager" output="screen"/>
	<node pkg="go_thesis" type="heron_controller.py" name="heron_control" output="screen"/>

	<!-- Long Baseline -->
	<node pkg="go_thesis" type="lbl.py" name="lbl_comm" output="screen"/>

	<!-- Start dvl for desistek -->
	<node pkg="go_thesis" type="desistek_dvl.py" name="read_dvl" output="screen"/>

	<!-- Start Trilateration -->
	<node pkg="go_thesis" type="trilateration.py" name="trilateration_estimation" output="screen"/>

	<!-- Start Depth Sensor -->
	<node pkg="go_thesis" type="desistek_depth.py" name="depth_sensor" output="screen"/>
</launch>